src/app/beer_garden/user.py

```diff
@@ -123,17 +126,35 @@ def user_synced_with_garden(user: User, garden: Garden) -> bool:
     # Avoiding circular imports
     from beer_garden.api.http.schemas.v1.user import UserSyncSchema
 
+    user = _filter_role_assigments_by_garden(user, garden)
+
     try:
         remote_user = RemoteUser.objects.get(username=user.username, garden=garden.name)
     except RemoteUser.DoesNotExist:
-        return False
+        return len(user.role_assignments) == 0
```

Comment:
As originally written this returned `False` if there was no `RemoteUser` entry, when actually there being no entry is the correct state if there are no relevant role assignments.